search/includes/classes/class-settingshealthjob.php

```diff
@@ -250,7 +249,7 @@ public function maybe_schedule_build_new_index( $indexable ) {
 			$this->send_alert( '#vip-go-es-alerts', $message, 2 );
 			
 			return;
-		} elseif ( ! wp_next_scheduled( self::CRON_EVENT_BUILD_NAME, [ $indexable ] ) ) {
+		} elseif ( ! wp_next_scheduled( self::CRON_EVENT_BUILD_NAME, [ $indexable->slug ] ) ) {
```

Comment:
Yes, it is intentional. It was an oversight on my part earlier in #2795, since I changed it to pass in the slug to the cron job so we didn't have to worry about any deserialization/serialization bugs.